src/GraphQL/GraphQLExtensions.cs

```diff
@@ -153,7 +153,9 @@ public static IGraphType BuildNamedType(this Type type, Func<Type, IGraphType> r
                        $"Expected non-null value, {nameof(resolve)} delegate return null for \"${type}\"");
         }
 
-        public static IEnumerable<string> IsValidLiteralValue(this IGraphType type, IValue valueAst, ISchema schema)
+        private static readonly string[] _foundNull = new[] { "Expected non-null value, found null" };
```

Comment:
I do not think that the ability to change the value of an array element should be considered a threat.